clippy_lints/src/methods/unnecessary_lazy_eval.rs

```diff
@@ -48,20 +48,16 @@ pub(super) fn check<'tcx>(
                     Applicability::MaybeIncorrect
                 };
 
-                span_lint_and_sugg(
-                    cx,
-                    UNNECESSARY_LAZY_EVALUATIONS,
-                    expr.span,
-                    msg,
-                    &format!("use `{}` instead", simplify_using),
-                    format!(
-                        "{0}.{1}({2})",
-                        snippet(cx, recv.span, ".."),
-                        simplify_using,
-                        snippet(cx, body_expr.span, ".."),
-                    ),
-                    applicability,
-                );
+                if let hir::ExprKind::MethodCall(_, _, span) = expr.kind {
```

Comment:
Not the best way, but I'd prefer not to change this

https://github.com/rust-lang/rust-clippy/blob/b83c6323c7d788fe4b6921b000324fb09945e12b/clippy_lints/src/methods/mod.rs#L2102

Reply:
Can the comment about it be added here?